config/config_test.go

```diff
@@ -70,3 +73,131 @@ func TestDetermineUseLiveData(t *testing.T) {
 		test.That(t, useLiveData, test.ShouldBeTrue)
 	})
 }
+
+// makeCfgService creates the simplest possible config that can pass validation.
+func makeCfgService() config.Service {
+	model := resource.NewDefaultModel(resource.ModelName("test"))
+	cfgService := config.Service{Name: "test", Type: "slam", Model: model}
+	cfgService.Attributes = make(map[string]interface{})
+	cfgService.Attributes["config_params"] = map[string]string{
+		"mode": "test mode",
+	}
+	cfgService.Attributes["data_dir"] = "path"
+	cfgService.Attributes["use_live_data"] = true
+	return cfgService
+}
+
+func TestNewAttrConf(t *testing.T) {
+	testCfgPath := "config_path"
+	logger := golog.NewTestLogger(t)
+
+	t.Run("Empty config", func(t *testing.T) {
+		model := resource.NewDefaultModel(resource.ModelName("test"))
+		cfgService := config.Service{Name: "test", Type: "slam", Model: model}
+		_, err := NewAttrConfig(cfgService)
+		test.That(t, err, test.ShouldBeError)
+	})
+
+	t.Run("Simplest valid config", func(t *testing.T) {
+		cfgService := makeCfgService()
+		_, err := NewAttrConfig(cfgService)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("Config without required fields", func(t *testing.T) {
+		requiredFields := []string{"data_dir", "use_live_data"}
+		for _, requiredField := range requiredFields {
+			logger.Debugf("Testing SLAM config without %s", requiredField)
+			cfgService := makeCfgService()
+			delete(cfgService.Attributes, requiredField)
+			_, err := NewAttrConfig(cfgService)
+			test.That(t, err, test.ShouldBeError, newError(utils.NewConfigValidationFieldRequiredError(testCfgPath, requiredField).Error()))
+		}
+	})
+
+	t.Run("Config with invalid parameter type", func(t *testing.T) {
+		cfgService := makeCfgService()
+		cfgService.Attributes["use_live_data"] = "true"
+		_, err := NewAttrConfig(cfgService)
+		test.That(t, err, test.ShouldBeError)
+		cfgService.Attributes["use_live_data"] = true
+		_, err = NewAttrConfig(cfgService)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	/* Unused config parameters should throw an error or fail. See: RSDK-2076
```

Comment:
I think it's better to leave this test out. We won't forget to add a test as part of RSDK-2076.